### PR TITLE
(DOCSP-21346): Revised the Prometheus tutorial.

### PR DESCRIPTION
**Pull request by corryroot**
@rodrigovalin requested a review of the new CKO Prometheus tutorial.

[JIRA](https://jira.mongodb.org/browse/DOCSP-21346)

### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: docs/prometheus/README.md [hunk 1]
```diff
@@ -1,114 +1,129 @@
...
-to indicate Prometheus how to consume metrics data from it.
+to indicate to Prometheus how to consume metrics data from 
+it.
 
-This is a simple MongoDB resource with one user, and with the `spec.Prometheus`
-attribute with basic HTTP Auth and no TLS, that will allow you to test
-Prometheus metrics coming from MongoDB.
+The sample specifies a simple MongoDB resource with one user,
+and the `spec.Prometheus` attribute with basic HTTP 
+authentication and no TLS. The sample will allow you to test
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:11]: avoid future tense https://docs.mongodb.com/meta/style-guide/writing/use-present-tense/
```suggestion
authentication and no TLS. The sample allows you to test
```

File: docs/prometheus/README.md [hunk 2]
```diff
@@ -1,114 +1,129 @@
...
+the Prometheus metrics coming from MongoDB.
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:12]: avoid passive voice constructions
```suggestion
metrics that MongoDB sends to Prometheus.
```

File: docs/prometheus/README.md [hunk 3]
```diff
@@ -1,114 +1,129 @@
...
-instructions
-[here](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack#kube-prometheus-stack):
+* Kubernetes 1.16+
+* Helm 3+
 
-This can be done with:
+### Install the Prometheus Operator
+
+You can install the Prometheus Operator using Helm. To learn 
+more, see the [installation instructions](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack#kube-prometheus-stack):
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:27]:
```suggestion
more, see the [installation instructions](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack#kube-prometheus-stack).
```

File: docs/prometheus/README.md [hunk 4]
```diff
@@ -1,114 +1,129 @@
...
+
+To install the Prometheus Operator using Helm, run the 
+following commands:
 
 ``` shell
 helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
 helm repo update
-helm install prometheus prometheus-community/kube-prometheus-stack --namespace prometheus-system --create-namespace
+helm install prometheus prometheus-community/kube-prometheus-stack --namespace <prometheus-system> --create-namespace
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:35]: **suggestion**: this is a lengthy command. consider using a `\` to split it to multiple lines. If you do, indent the second line by two spaces to make it clear that it's a continuation of the same command.

File: docs/prometheus/README.md [hunk 5]
```diff
@@ -1,114 +1,129 @@
...
+Operator and create a namespace to contain the Community 
+Kubernetes Operator and resources:
 
 ``` shell
-kubectl create namespace mongodb
+helm install community-operator mongodb/community-operator --namespace <mongodb> --create-namespace
 ```
 
-* Follow the [Installation Instructions](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/docs/install-upgrade.md#operator-in-same-namespace-as-resources)
+To learn more, see the [Installation Instructions](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/docs/install-upgrade.md#operator-in-same-namespace-as-resources).
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:48]: consider a relative link here as well

File: docs/prometheus/README.md [hunk 6]
```diff
@@ -1,114 +1,129 @@
...
 
-## Creating a MongoDB Resource
+## Create a MongoDB Resource
 
-We have created a sample yaml definition that you can use to create a MongoDB
-resource and a `ServiceMonitor` that will indicate Prometheus to start scraping
-its metrics information.
+ You can use the [mongodb-prometheus-sample.yaml](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/docs/prometheus/mongodb-prometheus-sample.yaml) file to
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:52]: **question**: it seems like this text is duplicated above. is this intended?

File: docs/prometheus/README.md [hunk 7]
```diff
@@ -1,114 +1,129 @@
...
-  --version v1.7.1 \
-  --set installCRDs=true
-```
+2. Now with Cert-Manager installed, create a Cert-Manager 
+   `Issuer` and then a `Certificate`. You can use the two files 
+   that we provide to create a new `Issuer`:
 
-Now with Cert-Manager installed we we'll create a Cert-Manager `Issuer` and then
-a `Certificate`. We provide 2 files that can be used to create a new `Issuer`.
+   a. Run the following command to create a `Secret` holding a
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:96]: disagreement between `a` and `entries`
```suggestion
   a. Run the following command to create a `Secret` that contains 
```

File: docs/prometheus/README.md [hunk 8]
```diff
@@ -1,114 +1,129 @@
...
+2. Now with Cert-Manager installed, create a Cert-Manager 
+   `Issuer` and then a `Certificate`. You can use the two files 
+   that we provide to create a new `Issuer`:
 
-Now with Cert-Manager installed we we'll create a Cert-Manager `Issuer` and then
-a `Certificate`. We provide 2 files that can be used to create a new `Issuer`.
+   a. Run the following command to create a `Secret` holding a 
+      TLS certificate `tls.crt` and `tls.key` entries. You can 
+      use the certificate and key files that we provide in the 
+      [`testdata/tls`](https://github.com/mongodb/mongodb-kubernetes-operator/tree/master/testdata/tls)
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:99]: consider a relative link

File: docs/prometheus/README.md [hunk 9]
```diff
@@ -1,114 +1,129 @@
...
+      TLS certificate `tls.crt` and `tls.key` entries. You can 
+      use the certificate and key files that we provide in the 
+      [`testdata/tls`](https://github.com/mongodb/mongodb-kubernetes-operator/tree/master/testdata/tls) 
+      directory to create a Cert-Manager `Certificate`.
 
-First we need to create a `Secret` holding a TLS certificate `tls.crt` and
-`tls.key` entries. We provide the certificate and key files that can be used to
-create a Cert-Manager `Certificate`, they are in the `testdata/tls` directory.
+      ``` shell
+      $ kubectl create secret tls issuer-secret --cert=../../testdata/tls/ca.crt --key=../../testdata/tls/ca.key \
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:103]: we don't include the shell prompt `$` in the examples above. I think you can remove it here.

File: docs/prometheus/README.md [hunk 10]
```diff
@@ -1,114 +1,129 @@
...
+          --namespace mongodb
+      secret/issuer-secret created
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:105]: I think this is the response for the command above. consider removing it or clearly indicating that it's not part of the command that you run to create a secret

File: docs/prometheus/README.md [hunk 11]
```diff
@@ -1,114 +1,129 @@
...
-And now we are ready to create a new `Issuer` and `Certificate`, by running the
-following command:
-
-``` shell
-$ kubectl apply -f issuer-and-cert.yaml --namespace mongodb
-issuer.cert-manager.io/ca-issuer created
-certificate.cert-manager.io/prometheus-target-cert created
-```
+      ``` shell
+      $ kubectl apply -f issuer-and-cert.yaml --namespace mongodb
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:112]: see comments on previous command

File: docs/prometheus/README.md [hunk 12]
```diff
@@ -1,114 +1,129 @@
...
-`CustomResource`; we can do it executing the following
-[patch](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/)
-operation.
+To enable TLS, you must add a new entry to the
+`spec.prometheus` section of the MongoDB `CustomResource`. Run 
+the following [patch](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/)
+operation to add the needed entry.
 
 ``` shell
 $ kubectl patch mdbc mongodb --type='json' \
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:129]: see comments above re: shell prompt and response

File: docs/prometheus/README.md [hunk 13]
```diff
@@ -118,13 +133,14 @@ $ kubectl patch mdbc mongodb --type='json' \
...
+`ServiceMonitor` to point to the HTTPS endpoint.
 
 ### Update ServiceMonitor
 
-To update our `ServiceMonitor` we will again patch the resource:
+To update the `ServiceMonitor`, run the following command to 
+patch the resource again:
 
 ``` shell
 $ kubectl patch servicemonitors mongodb-sm --type='json' \
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:146]: see comments above re: shelll prompt and response

File: docs/prometheus/README.md [hunk 14]
```diff
@@ -139,10 +155,11 @@ $ kubectl patch servicemonitors mongodb-sm --type='json' \
...
-Prometheus to not verify TLS certificates on MongoDB's end.
+With these changes, the new `ServiceMonitor` points to the HTTPS
+endpoint (defined in `/spec/endpoints/0/scheme`). You also 
+set `spec/endpoints/0/tlsConfig/insecureSkipVerify` to `true`, 
+so that Prometheus doesn't verify the TLS certificates on 
+MongoDB's end.
 
-Prometheus should now be able to scrape the MongoDB's target using HTTPS this
-time.
+Prometheus should now be able to scrape the MongoDB's target
```
Comment (jwilliams-mongo) [inline docs/prometheus/README.md:164]:
```suggestion
Prometheus should now be able to scrape the MongoDB target 
```

**Review by rodrigovalin — APPROVED** [review]
LGTM

File: docs/prometheus/README.md (hunk 6 above)
Comment (corryroot) [reply docs/prometheus/README.md:52]: yes

File: docs/prometheus/README.md [hunk 15]
```diff
@@ -1,114 +1,129 @@
...
+ deploy a MongoDB resource in your Kubernetes cluster, with a
+[`ServiceMonitor`](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md#related-resources)
+to indicate to Prometheus how to consume metrics data from 
+it.
 
-You can apply it directly with:
+You can apply the sample directly with the following command:
 
 ``` shell
 kubectl apply -f mongodb-prometheus-sample.yaml
```
[the remark this answers is not in the document]
Comment (corryroot) [reply docs/prometheus/README.md:61]: @rodrigovalin?

[the remark this answers is not in the document]
Comment (corryroot) [reply docs/prometheus/README.md:63]: @jwilliams-mongo, Rodrigo confirmed. I'll add a note.

**Review by jwilliams-mongo — APPROVED** [review]
@corryroot lgtm

**Review by irajdeep — APPROVED** [review]
LGTM

**Review by priyolahiri — APPROVED** [review]
LGTM

**Pull request merged**